token-metadata/program/src/state.rs

```diff
@@ -102,19 +107,19 @@ pub trait TokenMetadataAccount {
         Ok(())
     }
 
-    fn safe_deserialize<T: BorshDeserialize>(mut data: &[u8]) -> Result<T, BorshError> {
-        if !is_correct_account_type(data, Self::key(), Self::size()) {
+    fn safe_deserialize(mut data: &[u8]) -> Result<Self, BorshError> {
+        if !Self::is_correct_account_type(data, Self::key(), Self::size()) {
             return Err(BorshError::new(ErrorKind::Other, "DataTypeMismatch"));
         }
 
-        let result: T = T::deserialize(&mut data)?;
+        let result: Self = Self::deserialize(&mut data)?;
```

Comment:
Just wondering, does `let result = Self::deserialize(&mut data)?;` also work here  (and in the similar situation below)?